src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -477,7 +485,9 @@ protected function mutateAttributeForArray($key, $value)
      */
     protected function castAttribute($key, $value)
     {
-        if (is_null($value)) {
+        if ($this->isCustomCastable($key)) {
+            return $this->fromCustomCastable($key, $value);
+        } elseif (is_null($value)) {
             return $value;
         }
```

Comment:
Maybe we get rid of the `elseif`, in cases we `return` in both cases.

Reply:
Replaced by separate conditions